lib/babel.js

```diff
@@ -300,56 +20,83 @@ module.exports = (babel) => {
-          this.xlinkNamespaceId = path.scope.generateUidIdentifier('xlinkNamespace')
-        },
         exit (path, state) {
-          const appendChildModule = this.opts.appendChildModule || 'nanohtml/lib/append-child'
-          const setAttributeModule = this.opts.setAttributeModule || 'nanohtml/lib/set-attribute'
-          const useImport = this.opts.useImport
-
-          if (this.appendChildId.used) {
-            addImport(this.appendChildId, appendChildModule)
-          }
-          if (this.setAttributeId.used) {
-            addImport(this.setAttributeId, setAttributeModule)
-          }
-          if (this.svgNamespaceId.used) {
-            path.scope.push({
-              id: this.svgNamespaceId,
-              init: t.stringLiteral(SVGNS)
-            })
-          }
-          if (this.xlinkNamespaceId.used) {
-            path.scope.push({
-              id: this.xlinkNamespaceId,
-              init: t.stringLiteral(XLINKNS)
+          // Check every binding if it can be replaced with `createElement` implementation only
+          for (const binding of this.nanohtmlBindings) {
+            const replace = binding.referencePaths.every(function (reference) {
+              const node = reference.parentPath.node
+              if (!node) return true // node got removed
+              return t.isCallExpression(node) && node.callee.property.name === 'createElement'
             })
-          }
 
-          function addImport (id, source) {
-            if (useImport) {
-              path.unshiftContainer('body', t.importDeclaration([
-                t.importDefaultSpecifier(id)
-              ], t.stringLiteral(source)))
-            } else {
-              path.scope.push({
-                id: id,
-                init: t.callExpression(t.identifier('require'), [t.stringLiteral(source)])
-              })
+            if (replace) {
+              var library
+              var identifier = binding.identifier
+
+              if (binding.path.parentPath.isImportDeclaration()) {
+                library = binding.path.parentPath.node.source.value
+                binding.path.parentPath.remove()
+              } else {
+                library = binding.path.node.init.arguments[0].value
+                binding.path.remove()
+              }
+
+              if (this.opts.useImport) {
+                var createElement = path.scope.generateUidIdentifier('createElement')
+                path.scope.push({
+                  id: identifier,
+                  init: t.objectExpression([
+                    t.objectProperty(
+                      t.stringLiteral('createElement'),
+                      createElement
+                    )
+                  ])
+                })
+                path.unshiftContainer('body', t.importDeclaration([
+                  t.importDefaultSpecifier(createElement)
+                ], t.stringLiteral(library + '/lib/createElement')))
```

Comment:
I think we should just always output `nanohtml/lib/createElement` in the babel transform, only nanohtml exposes that. this means that transformed code will have a peer dependency on `nanohtml`, but that was also a limitation in the yo-yoify babel plugin.